build_tools/cmake/iree_tablegen_doc.cmake

```diff
@@ -60,11 +60,16 @@ function(iree_tablegen_doc)
   while(_RULE_OUTS)
     list(GET _RULE_OUTS 0 _COMMAND)
     list(REMOVE_AT _RULE_OUTS 0)
+    list(LENGTH _RULE_OUTS _LEN)
+    if(_LEN GREATER 1)
+      list(GET _RULE_OUTS 0 _DIALECT)
+      list(REMOVE_AT _RULE_OUTS 0)
+    endif()
     list(GET _RULE_OUTS 0 _OUTPUT)
     list(REMOVE_AT _RULE_OUTS 0)
 
     # TableGen this output with the given command.
-    tablegen(${_TBLGEN} ${_OUTPUT} ${_COMMAND} ${_INCLUDE_DIRS})
+    tablegen(${_TBLGEN} ${_OUTPUT} ${_COMMAND} ${_DIALECT} ${_INCLUDE_DIRS})
```

Comment:
I'm not a cmake expert. Is there a better way for doing it?

All the dialects except VMVX need to add `--dialect=xxx` flag. The _RULE_OUTS is a list of either `--gen-dialect-doc -dialect=flow FlowDialect.md` or `--gen-dialect-doc VMVXDialect.md`.

We'd like to pop the last element to `_OUTPUT` and pass the rest of elements as flags to tablegen.